x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CacheFile.java

```diff
@@ -249,41 +248,36 @@ private void ensureOpen() {
     }
 
     CompletableFuture<Integer> fetchRange(
-        long position,
+        Tuple<Long, Long> range,
         CheckedBiFunction<Long, Long, Integer, IOException> onRangeAvailable,
         CheckedBiConsumer<Long, Long, IOException> onRangeMissing
     ) {
         final CompletableFuture<Integer> future = new CompletableFuture<>();
         try {
-            if (position < 0 || position > tracker.getLength()) {
-                throw new IllegalArgumentException("Wrong read position [" + position + "]");
+            if (range.v1() < 0 || range.v1() > tracker.getLength() || range.v2() < 0 || range.v2() > tracker.getLength()) {
```

Comment:
Also invalid if the start of the range is after the end?

Reply:
Right, I pushed 4dc76b83c10f3881a7783896dd3610bc7baae522